src/crc16/simd.rs

```diff
@@ -0,0 +1,74 @@
+use crate::simd::{crc32_simd_coefficients, crc32_update_refin, SimdValue};
+use crate::table::crc16_table_slice_16;
+use crate::{Algorithm, Crc, Digest, Simd};
+
+use super::{finalize, init, update_slice16};
+
+impl Crc<Simd<u16>> {
+    pub const fn new(algorithm: &'static Algorithm<u16>) -> Self {
+        Self {
+            algorithm,
+            table: (
+                crc16_table_slice_16(algorithm.width, algorithm.poly, algorithm.refin),
+                unsafe {
+                    // SAFETY: Both represent numbers.
```

Comment:
Could the safety comment provide a bit more info? E.g: "This is safe since [u64; 8] has the same representation as [__m128i; 4]". Also, since this transmute is used in a couple places, I think it's worth moving to `x86.rs` with a safe encapsulation. Or just use [`bytemuck`](https://docs.rs/bytemuck).